src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/AgencyLocationEstablishment.kt

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.hmpps.prison.repository.jpa.model
+
+import jakarta.persistence.Embeddable
+import jakarta.persistence.EmbeddedId
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import org.hibernate.annotations.JoinColumnOrFormula
+import org.hibernate.annotations.JoinColumnsOrFormulas
+import org.hibernate.annotations.JoinFormula
+import java.io.Serializable
+
+@Embeddable
+data class AgencyLocationEstablishmentId(
+  @JoinColumn(name = "AGY_LOC_ID", nullable = false)
+  val agencyLocationId: String,
+
+  @JoinColumn(name = "ESTABLISHMENT_TYPE", nullable = false)
+  val establishmentTypeCode: String,
+) : Serializable
+
+@Entity
+@Table(name = "AGY_LOC_ESTABLISHMENTS")
+data class AgencyLocationEstablishment(
+  @EmbeddedId
+  val id: AgencyLocationEstablishmentId,
```

Comment:
Switch to `EmbeddedId` as the `IdClass` wasn't mapping correctly on a `OneToMany` annotation in the parent class `AgencyLocation`.

Reply:
Actually to be more accurate... 

We could have just continued to use `@Id` on the columns as long as we modelled `AgencyLocation` in the child as a Hibernate managed object instead of just the id. This would then fulfil the oppopsite side of the bidirectional relationship and Hibernate would be happy. (and note we could have removed the `@IdClass` as that's only needed for repositories but we don't have one for this class).

BUT as I changed the `establishmentType` to be a reference code this means that we had to use a `JoinFormula` and you can't have both `@Id` and a join formula on the same column (Hibernate throws in that case). Hence we need `@EmbeddedId`.